terraform/gt-site.tf

```diff
@@ -0,0 +1,91 @@
+provider "aws" {
+  profile = "${var.profile}" # Assumes you have valid AWS credentials.
+  region  = "${var.region}"
+}
+
+# --- ECS RESOURCES --- #
+
+resource "aws_ecs_cluster" "gt-site-cluster" {
+  name = "gt-site-cluster"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# `containers.json` is analogous to `docker-compose.yml`.
+# All the container settings are there.
+resource "aws_ecs_task_definition" "gt-site-cluster-task" {
+  family                = "gt-site-cluster"
+  container_definitions = "${file("containers.json")}"
+}
+
+resource "aws_ecs_service" "gt-site-cluster-service" {
+  name                               = "gt-site-cluster"
+  cluster                            = "${aws_ecs_cluster.gt-site-cluster.id}"
+  task_definition                    = "${aws_ecs_task_definition.gt-site-cluster-task.family}:${aws_ecs_task_definition.gt-site-cluster-task.revision}"
+  desired_count                      = "${var.desired_instance_count}"
+  iam_role                           = "${var.ecs_service_role}"
+  deployment_minimum_healthy_percent = "0"
+  deployment_maximum_percent         = "100"
+
+  load_balancer {
+    elb_name       = "${aws_elb.gt-site-elb.name}"
+    container_name = "gtsite-nginx"
+    container_port = 8080
+  }
+}
+
+# A load balancer is necessary to launch the EC2 manager instance properly.
+resource "aws_elb" "gt-site-elb" {
+  subnets = ["${var.subnet_id}"]
+
+  listener {
+    lb_port           = 80
+    lb_protocol       = "HTTP"
+    instance_port     = 8080   # The port that the nginx container is listening on.
+    instance_protocol = "HTTP"
+  }
+
+  cross_zone_load_balancing = false
+  idle_timeout              = 3600
+
+  tags {
+    Name = "gt-site-cluster elb"
+  }
+}
+
+resource "aws_launch_configuration" "ecs" {
+  name                 = "ECS ${aws_ecs_cluster.gt-site-cluster.name}"
+  image_id             = "${var.aws_ecs_ami}"
+  instance_type        = "${var.ec2_instance_type}"
+  iam_instance_profile = "${var.ecs_instance_profile}"
```

Comment:
Similar to the ECS Service IAM role, this instance profile IAM resource should be created within Terraform:

```hcl
data "aws_iam_policy_document" "container_instance_ec2_assume_role" {
  statement {
    effect = "Allow"

    principals {
      type        = "Service"
      identifiers = ["ec2.amazonaws.com"]
    }

    actions = ["sts:AssumeRole"]
  }
}

resource "aws_iam_role" "container_instance_ec2" {
  name               = "${var.environment}ContainerInstanceProfile"
  assume_role_policy = "${data.aws_iam_policy_document.container_instance_ec2_assume_role.json}"
}

resource "aws_iam_role_policy_attachment" "ecs_for_ec2_policy_container_instance_role" {
  role       = "${aws_iam_role.container_instance_ec2.name}"
  policy_arn = "${var.aws_ecs_for_ec2_service_role_policy_arn}"
}

resource "aws_iam_role_policy_attachment" "cloudwatch_logs_policy_container_instance_role" {
  role       = "${aws_iam_role.container_instance_ec2.name}"
  policy_arn = "${var.aws_cloudwatch_logs_policy_arn}"
}

resource "aws_iam_instance_profile" "container_instance" {
  name  = "${aws_iam_role.container_instance_ec2.name}"
  roles = ["${aws_iam_role.container_instance_ec2.name}"]
}
```

Again, here `aws_ecs_for_ec2_service_role_policy_arn` and `aws_cloudwatch_logs_policy_arn` end up being the following values:

- `arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role`
- `arn:aws:iam::aws:policy/CloudWatchLogsFullAccess`